api/v1alpha1/nodeobservabilitymachineconfig_helpers.go

```diff
@@ -119,3 +121,23 @@ func IsNodeObservabilityMachineConfigConditionPresentAndEqual(conditions []NodeO
 	}
 	return false
 }
+
+// IsNodeObservabilityMachineConfigConditionAsExpected returns true when conditionType is present and set to `ConditionTrue`, false when not present or set to `ConditionInProgress`. Returns error when anyother conditionType is set to `ConditionTrue` or `ConditionInProgress`
+func IsNodeObservabilityMachineConfigConditionAsExpected(conditions []NodeObservabilityMachineConfigCondition, conditionType NodeObservabilityMachineConfigConditionType) (bool, error) {
+	for _, cond := range conditions {
+		if cond.Type == conditionType {
```

Comment:
question: do we expect the matching condition to always be the first one?
Should we keep the result of the else L135 in a var till  after the loop in order to return false once all conditions were cheched?